File: backend/src/main/java/com/techlab/api/model/Producto.java
package com.techlab.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "tipo", discriminatorType = DiscriminatorType.STRING)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public abstract class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String nombre;

    @Min(0)
    @Column(nullable = false)
    private double precio;

    @Min(0)
    @Column(nullable = false)
    private int cantidadEnStock;

    public abstract String getTipo();

    public Producto(String nombre, double precio, int cantidadEnStock) {
        this.nombre = nombre;
        this.precio = precio;
        this.cantidadEnStock = cantidadEnStock;
    }
}

File: frontend/src/types/producto.ts
export type TipoProducto = 'BEBIDA' | 'COMIDA';

export interface Producto {
  id: number;
  tipo: TipoProducto;
  nombre: string;
  precio: number;
  cantidadEnStock: number;
}

export interface ProductoRequest {
  nombre: string;
  precio: number;
  cantidadEnStock: number;
  tipo: TipoProducto;
}
